rest/data/test_test.go

```diff
@@ -311,3 +317,138 @@ func TestFindTestsByDisplayTaskId(t *testing.T) {
 	assert.Error(err)
 	assert.Len(foundTests, 0)
 }
+
+func TestTestCountByTaskID(t *testing.T) {
+	require.NoError(t, db.ClearCollections(task.Collection, testresult.Collection))
+	defer func() {
+		assert.NoError(t, db.ClearCollections(task.Collection, testresult.Collection))
+	}()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Run("CedarTestResults", func(t *testing.T) {
+		regularTask := &task.Task{
+			Id:              "cedar",
+			HasCedarResults: true,
+		}
+		require.NoError(t, regularTask.Insert())
+		displayTask := &task.Task{
+			Id:              "cedar_display",
+			Execution:       1,
+			HasCedarResults: true,
+			DisplayOnly:     true,
+		}
+		require.NoError(t, displayTask.Insert())
+		handler := &mockCedarHandler{}
+		srv := httptest.NewServer(handler)
+		defer srv.Close()
+		evergreen.GetEnvironment().Settings().Cedar.BaseURL = srv.URL
+
+		for _, test := range []struct {
+			name       string
+			task       *task.Task
+			response   apimodels.CedarTestResultsStats
+			statusCode int
+			hasErr     bool
+		}{
+			{
+				name: "TaskDNE",
+				task: &task.Task{
+					Id:              "DNE",
+					HasCedarResults: true,
+				},
+				statusCode: http.StatusOK,
+				hasErr:     true,
+			},
+			{
+				name:       "CedarError",
+				task:       regularTask,
+				statusCode: http.StatusInternalServerError,
+				hasErr:     true,
+			},
+			{
+				name:       "TaskWithoutResults",
+				task:       regularTask,
+				statusCode: http.StatusNotFound,
+			},
+			{
+				name:       "TaskWithResults",
+				task:       regularTask,
+				response:   apimodels.CedarTestResultsStats{TotalCount: 100},
+				statusCode: http.StatusOK,
+			},
+			{
+				name:       "DisplayTask",
+				task:       displayTask,
+				response:   apimodels.CedarTestResultsStats{TotalCount: 100},
+				statusCode: http.StatusOK,
+			},
+		} {
+			t.Run(test.name, func(t *testing.T) {
+				responseData, err := json.Marshal(&test.response)
+				require.NoError(t, err)
+				handler.Response = responseData
+				handler.StatusCode = test.statusCode
+
+				count, err := TestCountByTaskID(ctx, test.task.Id, test.task.Execution)
```

Comment:
These seem to always assume that there's a single execution and it's the latest one. Should there be a test for other execution cases?

* Non-latest archived task executions
* Zero execution with multiple task executions
* Positive execution numbers that are higher than the latest execution

Reply:
I think that is testing too much of the Cedar specific behavior, which this test can assume just works. We really just want to make sure we are passing in the right parameters to Cedar's API endpoint and that we handle the Cedar response accordingly. I can add a test that checks that this function fails if the execution does not exist (that would come from the database check before hitting the API endpoint).

Reply:
I added a test to make sure that it will fail when the execution differs from the one passed in to the function.